addons/analytic/models/ir_config_parameter.py

```diff
@@ -0,0 +1,32 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, _
+from odoo.exceptions import UserError
+
+
+class IrConfigParameter(models.Model):
+
+    '''
+    Dynamic fields are created for all analytic plans except the analytic plan associated with the analytic.project_plan system paramater.
+    When this paramater is changed a dynamic field needs to be generated for the plan that was associated with this parameter
+    and the dynamic field associated with the value of the new parameter needs to be removed.
+    '''
+
+    _inherit = 'ir.config_parameter'
+
+    def write(self, vals):
+        plan_id = False
+        plan_field = False
+
+        for rec in self:
+            if rec.key == 'analytic.project_plan' and rec.value != vals['value']:
+                plan_id = rec.value
+                plan_field = self.env['account.analytic.plan'].browse(int(vals['value']))._find_plan_column()
```

Comment:
- int('') and int(None) blow up with a traceback, so you want to guard about invalid / empty / deleted values before calling `int` (or find a safer alternative)